pages/posts/why-portal-needs-to-be-its-own-network.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Why Portal Needs To Be Its Own Network
+date: 2024/03/12
+description: Explaining why the Portal Network needed to build its own network rather than leverage an existing distributed storage engine like Filecoin or Swarm.
+tag: project, Ethereum, p2p, storage
+author: Piper Merriam
+---
+
+We commonly get asked why we don't use one of the various general purpose
+distributed storage networks (like Filecoin/IPFS).  Baked into this question is
+an assumption that general purpose storage networks would solve the problem
+that Portal is aiming to solve.
+
+## Problem 1: Content Addressing
+
+One issue that arises with general purpose storage networks is around content
+addressing.  We are not able to use just any arbitrary content addressing
+scheme.  Portal Network requires that all data be verifiable as canonical chain
+data.  The way Ethereum data is verified is baked into the Ethereum protocol.
+A simple example of this is verification of a block header.  This involves
+taking the `keccak` hash of the RLP encoded block header.  A more complex
+example of this is anchoring state data using a Hexary Merkle Patricia Trie
+proof to the `Header.state_root`.
```

Comment:
I'm not sure I understand here the direct relation that is made between addressing scheme and having verifiable data. 

Wouldn't verifiable data also be possible on general purpose network (of course, not mandatory on the basic layer; "build-in" as we do for Portal)?
Is the benefit of the custom content addressing not rather that it is more efficient and directly related to the application using it (Like, in general purpose solutions I assume we would a have to build some extra mapping constructions around it)?

Reply:
To be clear, I do understand that there is an indirect relation, but that one is basically explained in the next chapter: Competition for Storage

Reply:
We use multiple different methods to "address" our content.

- Block headers are addressed by `keccak(rlp(header))`
- Block bodies are addressed by `hexary_merkle_trie(receipts)` (along with uncles)

I suspect that we have some SSZ based content addressing in the beacon network.

The idea is that to use an off-the-shelf storage network, that network would need to support all of these cryptographic addressing schemes.